prov/gni/src/gnix_datagram.c

```diff
@@ -334,41 +339,54 @@ int _gnix_dgram_bnd_post(struct gnix_datagram *d, gni_return_t *status_ptr)
 		goto err;
 	}
 
-	/*
-	 * if we get GNI_RC_ERROR_RESOURCE status return from
-	 * GNI_EpPostDataWId  that means that a previously posted wildcard
-	 * datagram has matched up with an incoming connect
-	 * request from the rank we are trying to send a connect
-	 * request to.  Don't treat this case as an error.
-	 */
-
 	fastlock_acquire(&d->nic->lock);
```

Comment:
Do you really have to lock before the pre-post callback?  I see below that you unlock before the post-post callback.